File: android/app/src/main/java/com/lena/grocery/model/GroceryList.kt
package com.lena.grocery.model

data class GroceryList(
    val groceryListID: Int,
    val mealPlanID: Int?,
    val generatedDate: String?,
    val groceryListItems: List<GroceryListItem>
)

data class GroceryListItem(
    val groceryListItemID: Int,
    val groceryListID: Int,
    val itemID: Int?,
    val itemName: String?,
    val manualItemName: String?,
    val quantityNeeded: Double,
    val unitOfMeasure: String?,
    val source: String,
    var isChecked: Boolean
)

File: LENA/Entity/Grocery/GroceryList.cs
using System;
using System.Collections.Generic;

using LENA.Domain.Entity.Common;

namespace LENA.Domain.Entity.Grocery
{
    public class GroceryList : AuditableEntity
    {
        public int GroceryListID { get; set; }
        public int UserID { get; set; }
        public int? MealPlanID { get; set; }
        public DateTime GeneratedDate { get; set; }

        public ICollection<GroceryListItem>? GroceryListItems { get; set; }
    }
}